src/generate/formats/writer/yaml/index.ts

```diff
@@ -65,22 +62,18 @@ export async function writeIntegration(
 
 export async function writePackageJson(
   name: string,
-  usedComponents: UsedComponents,
-  registryPrefix?: string,
+  usedComponents: Record<string, UsedComponent>,
 ) {
   const manifests: Record<string, string> = {};
 
-  usedComponents.public.forEach((componentKey) => {
-    if (!EXCLUDED_PUBLIC_COMPONENTS.includes(componentKey)) {
-      const packageName = `@component-manifests/${componentKey}`;
+  for (const [key, value] of Object.entries(usedComponents)) {
+    if (!EXCLUDED_PUBLIC_COMPONENTS.includes(key)) {
+      const packageName = `${value.registryPrefix}/${key}`;
+      // @TODO: We have access to the version here, we can eventually
+      // attempt to resolve the package version for them here.
```

Comment:
FYI that we'd need the `signature` to do so which will involve some YAML export changes to opt into `signature` vs `version` (and `LATEST`).